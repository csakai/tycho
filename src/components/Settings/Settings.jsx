import React from 'react';
import cx from 'classnames';
import ScaleSlider from '../Slider/ScaleSlider';
import Constants from '../../constants';

export default class Settings extends React.Component {
    render() {
        return (
            <div className={`settings-panel ${cx({
                'settings-panel--open': this.props.settingsActive
            })}`}>
                <div className="settings-panel__header">
                    <div className="settings-panel__title">
                        {this.props.pageText.settings}
                    </div>
                    <div
                        className={`settings-panel__hamburger ${cx({
                            'settings-panel__hamburger--open': this.props.settingsActive
                        })}`}
                        onClick={this.props.toggleSetting}>
                        <span></span>
                        <span></span>
                        <span></span>
                    </div>
                </div>
        
                <div className={`settings-panel__content ${cx({
                    'settings-panel__content--open': this.props.settingsActive
                })}`}>
                    <ScaleSlider
                        value={this.props.speed}
                        label={this.props.pageText.speedScale}
                        onChange={this.props.changeSpeed}
                        min={Constants.UI.Sliders.Speed.MIN}
                        max={Constants.UI.Sliders.Speed.MAX}
                    />
                    <br />
                    <ScaleSlider
                        value={this.props.scale}
                        label={this.props.pageText.planetScale}
                        onChange={this.props.changeScale}
                        min={Constants.UI.Sliders.Scale.MIN}
                        max={Constants.UI.Sliders.Scale.MAX}
                    />
                    <br />
                    <br />
                </div>
            </div>
        );
    }
}