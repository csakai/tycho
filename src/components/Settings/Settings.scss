.settings-panel {
  min-width: 300px;
  color: #fff;
  transition: 1s all;
  pointer-events: none;
  
  &__header {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: 100%;
  }
  
  &__title {
    align-items: center;
    flex: 1;
    display: flex;
    align-items: center;
    padding: 0.75em;
    opacity: 0;
    transition: all 1s;
    font-size: 1.5em;
  }
  
  &__content {
    padding: 0 1em;
    max-height: 0;
    overflow: hidden;
    transition: all 1s;
  }
  
  &__hamburger {
    width: 32px;
    height: 32px;
    position: relative;
    margin: 1em;
    transform: rotate(0deg);
    transition: 1s ease-in-out;
    cursor: pointer;
    pointer-events: auto;

    span {
      display: block;
      position: absolute;
      height: 4px;
      width: 100%;
      background: #fff;
      opacity: 1;
      left: 0;
      transform: rotate(0deg);
      transition: 0.5s ease-in-out;

      &:nth-child(1) {
        top: 3px;
        transform-origin: left center;
      }

      &:nth-child(2) {
        top: 15px;
        transform-origin: left center;
      }

      &:nth-child(3) {
        top: 27px;
        transform-origin: left center;
      }
    }
    
    &:hover span {
      box-shadow: 1px 1px 10px #fff;
    }

    &--open span {
      &:nth-child(1) {
        transform: rotate(45deg);
        top: 2px;
        left: 4px;
      }

      &:nth-child(2) {
        width: 0%;
        opacity: 0;
      }

      &:nth-child(3) {
        transform: rotate(-45deg);
        top: 24px;
        left: 4px;
      }
    }
  }
  
  &--open {
    background-color: rgba(0, 0, 0, 0.8);
    pointer-events: auto;
    
    .settings-panel__content {
      max-height: 200px;
    }
    
    .settings-panel__title {
      display: block;
      opacity: 1;
    }
  }
}