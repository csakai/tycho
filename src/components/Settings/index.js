import './Settings.css';

export {default} from './Settings';