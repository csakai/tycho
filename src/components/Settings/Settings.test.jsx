import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Settings from './Settings';

describe('Settings Component', () => {
    describe('render()', () => {    
        it('should render the settings component', () => {
            const component = shallow(<Settings pageText={{}} />);
      
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
