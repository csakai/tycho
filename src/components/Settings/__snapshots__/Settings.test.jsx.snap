// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component render() should render the settings component 1`] = `
<div
  className="settings-panel "
>
  <div
    className="settings-panel__header"
  >
    <div
      className="settings-panel__title"
    />
    <div
      className="settings-panel__hamburger "
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="settings-panel__content "
  >
    <ScaleSlider
      max={10}
      min={0}
    />
    <br />
    <ScaleSlider
      max={10}
      min={1}
    />
    <br />
    <br />
  </div>
</div>
`;
